hpecp/role.py

```diff
@@ -0,0 +1,158 @@
+from .exceptions import (
+    ContainerPlatformClientException,
+    APIException,
+    APIItemNotFoundException,
+    APIItemConflictException,
+)
+
+import json
+from operator import attrgetter
+
+# from tabulate import tabulate
+# from enum import Enum
+# import polling
+import re
+import six
+import sys
+
+try:
+    basestring
+except NameError:
+    basestring = str
+
+
+class RoleController:
+    """This is the main class that users will interact with to work with roles.
+
+    An instance of this class is available in the client.ContainerPlatformClient with the attribute name
+    :py:attr:`role <.client.ContainerPlatformClient.role>`.  The methods of this class can be
+    invoked using `client.role.method()`.  See the example below:
+
+    Example::
+
+        client = ContainerPlatformClient(...).create_session()
+        client.role.get()
+
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def get(self, role_id):
+        """Retrieve a Role by ID.
+
+        Args:
+            role_id: str
+                The role ID - format: '/api/v1/role/[0-9]+'
+
+        Returns:
+            Role: object representing Role
+
+        Raises:
+            APIException
+        """
+        assert isinstance(
+            role_id, str
+        ), "'role_id' must be provided and must be a string"
+        assert re.match(
+            r"\/api\/v1\/role\/[0-9]+", role_id
+        ), "'role_id' must have format '/api/v1/role/[0-9]+'"
+
+        response = self.client._request(
+            url=role_id, http_method="get", description="role/get"
+        )
+        if response.json()["purpose"] != "proxy":
+            raise APIItemNotFoundException(
+                message="role not found with id: " + role_id,
+                request_method="get",
+                request_url=role_id,
+            )
+
+        return Role(response.json())
+
+
+class Role:
+    """Create an instance of Role from json data returned from the HPE Container Platform API.
+
+    Users of this library are not expected to create an instance of this class.
+
+    Parameters:
+        json : str
+            The json returned by the API representing a Gateway.
+
+    Returns:
+        Gateway:
+            An instance of Gateway
+    """
+
+    all_fields = [
+        "id",
+        "name",
+        "description",
+    ]
+    """All of the fields of Role objects as returned by the HPE Container Platform API"""
+
+    default_display_fields = [
+        "id",
+        "name",
+        "description",
+    ]
+    """These fields are displayed by default, e.g. in tabulate()"""
+
+    def __init__(self, json):
+        self.json = json
+        self.display_columns = Role.default_display_fields
+
+    def __repr__(self):
+        return "<Role id:{} description:{}>".format(self.id, self.description)
+
+    def __str__(self):
+        return "K8sCluster(id={}, description={})".format(
+            self.id, self.description
+        )
```

Comment:
```suggestion
    def __str__(self):
        return "Role(id={}, description={})".format(
            self.id, self.description
        )
```

Reply:
Thanks for the correction. Was not sure about this one here.
Appreciate it :")